tests/MongoDB.Driver.Tests/PipelineDefinitionBuilderTests.cs

```diff
@@ -114,6 +114,61 @@ public void Merge_should_add_expected_stage()
             stages[0].Should().Be("{ $merge : { into : { db : 'database', coll : 'collection' } } }");
         }
 
+        [Fact]
+        public void UnionWith_should_add_expected_stage()
+        {
+            var pipeline = new EmptyPipelineDefinition<BsonDocument>();
+            var withCollection = Mock.Of<IMongoCollection<BsonDocument>>(
+                coll => coll.CollectionNamespace == CollectionNamespace.FromFullName("db.test"));
+            var withPipeline = new EmptyPipelineDefinition<BsonDocument>()
+                .AppendStage<BsonDocument, BsonDocument, BsonDocument>("{ $match : { b : 1 } }");
+
+            var result = pipeline.UnionWith<BsonDocument, BsonDocument, BsonDocument>(withCollection, withPipeline);
```

Comment:
Simplify to:

    var result = pipeline.UnionWith(withCollection, withPipeline);

let compiler infer generic type arguments.

Reply:
done